File: http-streams/src/main/java/rs/async/Asynchronous.java
package rs.async;

import java.util.List;

import akka.actor.AbstractActor;
import akka.actor.ActorRef;

import static akka.actor.ActorRef.noSender;

public class Asynchronous {
  static class Task {}
  static class Result {}

  //#actor
  static class MyActor extends AbstractActor {
    @Override
    public Receive createReceive() {
      return receiveBuilder()
              .match(Task.class, task -> { /* do work */ })
              .build();
    }
  }
  //#actor

  public void performTasks() {
    ActorRef actor = null;
    List<Task> tasks = null;

    //#enqueue
    for (Task task: tasks) {
      actor.tell(task, noSender());
    }
    //#enqueue
  }
}

File: http-streams/src/main/scala/rs/async/Asynchronous.scala
package sc.rs.async

import akka.actor.{Actor, ActorRef}

object Asynchronous {

  case class Task()

  private[async] class Result {}

  //#actor
  class MyActor extends Actor {
    override def receive = {
      case Task(/*params*/) ⇒ /* do work */
    }
  }
  //#actor

  def performTasks(): Unit = {
    val actor: ActorRef = null
    val tasks: Seq[Task] = null
    //#enqueue
    for (task <- tasks) {
      actor ! task
    }
    //#enqueue
  }
}
